File: src/main/scala/fi/vm/sade/hakuperusteet/domain/Payment.scala
package fi.vm.sade.hakuperusteet.domain

import java.util.Date

import fi.vm.sade.hakuperusteet.domain.PaymentStatus.PaymentStatus

case class Payment(id: Option[Int], personOid: String, timestamp: Date, reference: String, orderNumber: String, paymCallId: String, status: PaymentStatus, hakemusOid: Option[String])

object PaymentStatus extends Enumeration {
  type PaymentStatus = Value
  val started, ok, cancel, error = Value
}

File: src/main/assets-admin/payment/Payment.js
import Bacon from 'baconjs'

import HttpUtil from '../../assets/util/HttpUtil.js'
import {enableSubmitAndHideBusy} from '../../assets/util/HtmlUtils.js'
import {tarjontaForHakukohdeOid} from "../../assets/util/TarjontaUtil.js"

export function submitPaymentDataToServer(state, payment, form) {
  const paymentData = {
    id: payment.id,
    orderNumber: payment.orderNumber,
    paymCallId: payment.paymCallId,
    personOid: payment.personOid,
    reference: payment.reference,
    status: payment.status,
    timestamp: payment.timestamp
  }
  const promise = Bacon.fromPromise(HttpUtil.post(state.paymentUpdateUrl, paymentData))
  promise.onError((error) => {
    enableSubmitAndHideBusy(form)
    form.querySelector("span.general").classList.remove("hide")
  })
  return promise
}